src/main/java/org/opensearch/neuralsearch/search/util/TopDocsMerger.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.RequiredArgsConstructor;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TotalHits;
+import org.opensearch.common.lucene.search.TopDocsAndMaxScore;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+/**
+ * Utility class for merging TopDocs and MaxScore across multiple search queries
+ */
+@RequiredArgsConstructor
+public class TopDocsMerger {
```

Comment:
nit: HybridQueryTopDocsMerger. This class is exclusive for Hybrid Query.

Reply:
it's actually generic, there isn't any special logic for hybrid query. Will leave name as it's now

Reply:
The merge method is exclusive for hybrid query.

Reply:
Not really, it merges two TopDocsAndMaxScore object, only specific of hybrid query is in ScoreDocs merger.

Reply:
+1 to @martin-gaievski